scanspec/service.py

```diff
@@ -17,39 +26,30 @@
 class Points:
     """ A collection of singular or multidimensional locations in scan space"""
 
-    def __init__(self, points: Optional[ndarray]):
+    def __init__(self, points: ndarray):
         self._points = points
 
     @resolver
-    def string(self) -> Optional[str]:
+    def string(self) -> str:
         return array2string(self._points)
 
     @resolver
-    def float_list(self) -> Optional[List[float]]:
-        if self._points is None:
-            return None
-        else:
-            return self._points.tolist()
+    def float_list(self) -> List[float]:
+        return self._points.tolist()
 
     @resolver
-    def b64(self) -> Optional[str]:
-        if self._points is None:
-            return None
-        else:
-            # make sure the data is sent as float64
-            assert dtype(self._points[0]) == dtype(float64)
-            return base64.b64encode(self._points.tobytes()).decode("utf-8")
+    def b64(self) -> str:
+        # make sure the data is sent as float64
+        assert dtype(self._points[0]) == dtype(float64)
+        return base64.b64encode(self._points.tobytes()).decode("utf-8")
```

Comment:
```suggestion
        return base64.b64encode(self._points).decode()
```

Reply:
.tobytes() is needed to stop mypy complaining: Argument 1 to "b64encode" has incompatible type "ndarray"; expected "bytes"

Reply:
Fair enough, performance doesn't seem affected much so that's fine:
```
$ python -m timeit -s "import numpy as np; import base64; a = np.arange(100000)" "base64.b64encode(a.tobytes())"
200 loops, best of 5: 1.69 msec per loop
$ python -m timeit -s "import numpy as np; import base64; a = np.arange(100000)" "base64.b64encode(a)"
200 loops, best of 5: 1.68 msec per loop
```
If performance was affected we could add `# type: ignore` to it